src/fides/api/ops/models/registration.py

```diff
@@ -0,0 +1,40 @@
+import logging
+from typing import Any, Dict
+
+from fideslib.db.base_class import Base, FidesBase
+from sqlalchemy import (
+    Boolean,
+    Column,
+    String,
+)
+from sqlalchemy.orm import Session
+
+logger = logging.getLogger(__name__)
+
+
+class UserRegistration(Base):
+    """
+    Stores registration status of a particular Fides deployment.
+    """
+
+    user_email = Column(String, nullable=True)
+    user_organization = Column(String, nullable=True)
+    analytics_id = Column(String, unique=True, nullable=False)
+    opt_in = Column(Boolean, nullable=False, default=False)
+
+    @classmethod
+    def create_or_update(cls, db: Session, *, data: Dict[str, Any]) -> FidesBase:
+        """
+        Creates a registration if none exists, or updates an existing
+        registration matched by `analytics_id`.
+        """
+        existing_model = cls.get_by(
+            db=db,
+            field="analytics_id",
+            value=data["analytics_id"],
+        )
+        if existing_model:
+            existing_model.opt_in = data["opt_in"]
```

Comment:
> The function signature requires a caller to "just know" that the data dictionary must include an "opt_in" key in order for an update to have any effect

Great catch! I've actually changed the method to only update the keys passed in `data`, as per your other suggestion, so presence `opt_in` needn't be asserted since a caller may want to only update `user_email` etc.s